lib/ansible/modules/network/aci/aci_rest.py

```diff
@@ -0,0 +1,395 @@
+status_code:
+  description: HTTP status code
+  returned: always
+  type: int
+  sample: 400
+totalCount:
+  description: Number of items in the imdata array
+  returned: always
+  type: string
+  sample: '0'
+'''
+
+import json
+import os
+
+# Optional, only used for XML payload
+try:
+    import lxml.etree
+    HAS_LXML_ETREE = True
+except ImportError:
+    HAS_LXML_ETREE = False
+
+# Optional, only used for XML payload
+try:
+    from xmljson import cobra
+    HAS_XMLJSON_COBRA = True
+except ImportError:
+    HAS_XMLJSON_COBRA = False
+
+# from ansible.module_utils.aci import aci_login
+from ansible.module_utils.basic import AnsibleModule, get_exception
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_bytes
+
+
+aci_argument_spec = dict(
+    hostname=dict(type='str', required=True, aliases=['host']),
+    username=dict(type='str', default='admin', aliases=['user']),
+    password=dict(type='str', required=True, no_log=True),
+    protocol=dict(type='str'),  # Deprecated in v2.8
+    timeout=dict(type='int', default=30),
+    use_ssl=dict(type='bool', default=True),
+    validate_certs=dict(type='bool', default=True),
+)
+
+
+def aci_login(module, result=dict()):
+    ''' Log in to APIC '''
+
+    # Set protocol based on use_ssl parameter
+    if module.params['protocol'] is None:
+        module.params['protocol'] = 'https' if module.params.get('use_ssl', True) else 'http'
+
+    # Perform login request
+    url = '%(protocol)s://%(host)s/api/aaaLogin.json' % module.params
+    data = {'aaaUser': {'attributes': {'name': module.params['username'], 'pwd': module.params['password']}}}
+    resp, auth = fetch_url(module, url, data=json.dumps(data), method="POST", timeout=module.params['timeout'])
+
+    # Handle APIC response
+    if auth['status'] != 200:
+        try:
+            result.update(aci_response(auth['body'], 'json'))
+            result['msg'] = 'Authentication failed: %(error_code)s %(error_text)s' % result
+        except KeyError:
+            result['msg'] = '%(msg)s for %(url)s' % auth
+        result['response'] = auth['msg']
+        result['status_code'] = auth['status']
+        module.fail_json(**result)
+
+    return resp
+
+
+def aci_response(rawoutput, rest_type='xml'):
+    ''' Handle APIC response output '''
+
+    result = dict()
+
+    if rest_type == 'json':
+        # Use APIC response as module output
+        try:
+            result = json.loads(rawoutput)
+        except:
+            e = get_exception()
+            # Expose RAW output for troubleshooting
+            result['error_code'] = -1
+            result['error_text'] = "Unable to parse output as JSON, see 'raw' output. %s" % e
+            result['raw'] = rawoutput
+            return result
+    else:
+        # NOTE: The XML-to-JSON conversion is using the "Cobra" convention
+        xmldata = None
+        try:
+            xml = lxml.etree.fromstring(to_bytes(rawoutput))
```

Comment:
Can the native ElementTree api be used here.
```from xml.etree.ElementTree import fromstring```

This will reduce module dependency on `lxml`.